### ISSUE TITLE
🐛 BUG: D1 Platform Proxy Socket Closed (Probably Due To Timeout)

### ISSUE THREAD

**Issue by naourass** [cloudflare/workers-sdk#7376]
### Which Cloudflare product(s) does this pertain to?

D1, Wrangler, Miniflare

### What version(s) of the tool(s) are you using?

3.91.0 [Wrangler]

### What version of Node are you using?

Tested on 18 LTS, 20 LTS, and 22 LTS

### What operating system and version are you using?

Ubuntu 22.04.3 WSL2 on Windows 11

### Observed behavior
The proxy connection for local D1 binding simulation of `getPlatformProxy()` is closing automatically after a short period of time.

### Expected behavior
I expect the connection to close only after I call `dispose()` method of the platform proxy object, or at least after a fairly long period of inactivity.

### Steps to reproduce
```js
const cfPlatformProxy = await getPlatformProxy();
const db = drizzle(cfPlatformProxy.env.DB);

// Do some a few seconds long computation here...

const res = await db.select().from(myTable);

// You'll get `SocketError: other side closed` when trying to access the db now.
```

### Please provide any relevant error logs

```
/home/naourass/web_exp/phoenix-astro-next/node_modules/undici/index.js:112
        Error.captureStackTrace(err, this)
              ^

TypeError: fetch failed
    at Object.fetch (/home/naourass/web_exp/phoenix-astro-next/node_modules/undici/index.js:112:15)
    at fetch3 (/home/naourass/web_exp/phoenix-astro-next/node_modules/miniflare/src/http/fetch.ts:90:19)
    at ProxyClientBridge.dispatchFetch (/home/naourass/web_exp/phoenix-astro-next/node_modules/miniflare/src/index.ts:1647:20)
    at ProxyStubHandler.#parseAsyncResponse (/home/naourass/web_exp/phoenix-astro-next/node_modules/miniflare/src/plugins/core/proxy/client.ts:339:15)
    at D1PreparedQuery.get (/home/naourass/web_exp/phoenix-astro-next/node_modules/src/d1/session.ts:223:16) {
  cause: SocketError: other side closed
      at Socket.onSocketEnd (/home/naourass/web_exp/phoenix-astro-next/node_modules/undici/lib/client.js:1118:22)
      at Socket.emit (node:events:529:35)
      at Socket.emit (node:domain:489:12)
      at endReadableNT (node:internal/streams/readable:1400:12)
      at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
    code: 'UND_ERR_SOCKET',
    socket: {
      localAddress: '127.0.0.1',
      localPort: 57506,
      remoteAddress: undefined,
      remotePort: undefined,
      remoteFamily: undefined,
      timeout: undefined,
      bytesWritten: 1091,
      bytesRead: 1783105
    }
  }
}
```

**Comment by petebacondarwin** [comment]
Thanks for reporting this. I was able to reproduce the underlying failure on Wrangler 3.91.0 / Miniflare 3.20241106.1: after priming local D1 and synchronously blocking the Node.js event loop for 5 seconds, the next direct D1 query failed with TypeError: fetch failed caused by ECONNRESET. The same sequence through Drizzle failed at the query as reported.

I could not reproduce this with the current workspace bundle, Wrangler 4.123.0, on Node.js 22 or 24, including asynchronous idle periods up to 90 seconds and event-loop-blocking periods up to 30 seconds.

This appears to have been fixed indirectly by #11642, which prevents reuse of idle sockets closed by workerd. That fix shipped in Miniflare 4.20260103.0 and Wrangler 4.57.0.

Closing as fixed. If this recurs on a current Wrangler release, please open a new issue with an updated reproduction.
